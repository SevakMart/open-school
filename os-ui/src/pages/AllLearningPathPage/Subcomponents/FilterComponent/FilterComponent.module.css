.mainContainer {
  width: 358px;
  background-color: #ffffff;
}
.mainContainer .filterMainContent {
  border-radius: 18px;
  margin: 64px 26px 64px 48px;
}
.mainContainer .filterMainContent .mainTitle {
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 600;
  font-size: 28px;
  line-height: 24px;
  color: #222427;
  margin: 0;
}/*# sourceMappingURL=FilterComponent.module.css.map */