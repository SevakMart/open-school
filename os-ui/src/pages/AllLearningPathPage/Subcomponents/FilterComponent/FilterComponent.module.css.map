{"version":3,"sources":["FilterComponent.module.scss","../../../../styles/Configs/Colors.scss","FilterComponent.module.css","../../../../styles/Mixins/FontMixin.scss","../../../../styles/Configs/Fonts.scss"],"names":[],"mappings":"AAKA;EACI,YAAA;EACA,yBCHe;ACDnB;AFKI;EACI,mBAAA;EACA,2BAAA;AEHR;AFIQ;EGVN,yBCDY;EDEZ,kBHU2C;EGT3C,gBHSkD;EGRlD,eHQsD;EGPtD,iBHO2D;EACjD,cCVQ;EDWR,SAAA;AEEZ","file":"FilterComponent.module.css"}