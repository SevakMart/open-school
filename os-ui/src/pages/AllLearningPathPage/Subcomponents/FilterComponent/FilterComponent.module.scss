@import "../../../../styles/Mixins/FlexMixins.scss";
@import "../../../../styles/Mixins/FontMixin.scss";

.mainContainer,.hiddenContainer{
    @include flexStyle(row,center,space-between);
    width:15%;
    align-self:flex-start;
    .filterMainContent{
        background-color:#FFFFFF;
        border-radius:1.125rem;
        padding-left:0.5rem;
        width:100%;
        .mainTitle{
            @include fontStyle('Montserrat',normal,600,1.25rem,1.5rem);
            color:#222427;
        }
    }

}