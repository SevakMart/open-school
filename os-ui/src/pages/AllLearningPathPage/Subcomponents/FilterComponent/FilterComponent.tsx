import { useState, useEffect, useContext } from 'react';
import { tokenContext, courseContentContext } from '../../../../contexts/Contexts';
import courseService from '../../../../services/courseService';
import FilteringContent from '../FilteringContent/FilteringContent';
import { FilteringFeatureType } from '../../../../types/FilteringFeaturesType';
import { FILTER } from '../../../../constants/Strings';
import { CourseContent } from '../../../../types/CourseContent';
import styles from './FilterComponent.module.scss';

enum VisibleFilterTab {
    isVisible='visible',
    isHidden='hidden'
}

const FilterComponent = ({ changeVisibility }:{changeVisibility:()=>void}) => {
  const [visibleFilterTab, setVisibleFilterTab] = useState(VisibleFilterTab.isVisible);
  const [filterFeatures, setFilterFeatures] = useState<FilteringFeatureType>({});
  const token = useContext(tokenContext);
  const contentType = useContext(courseContentContext);
  const {
    mainContainer, hiddenContainer, filterMainContent, mainTitle, visibilityButton,
  } = styles;

  const toggleFilterTabVisibility = () => {
    if (visibleFilterTab === VisibleFilterTab.isVisible) {
      setVisibleFilterTab(VisibleFilterTab.isHidden);
      changeVisibility();
    } else {
      setVisibleFilterTab(VisibleFilterTab.isVisible);
      changeVisibility();
    }
  };

  useEffect(() => {
    courseService.getFilterFeatures({}, token).then((data) => {
      setFilterFeatures(data);
    });
  }, []);

  return (
    <div
      className={visibleFilterTab === 'visible' ? mainContainer
        : hiddenContainer}
      style={contentType === CourseContent.SAVEDCOURSES ? { display: 'none' } : { display: 'flex' }}
    >
      <div className={filterMainContent}>
        <p className={mainTitle}>{FILTER}</p>
        {
          Object.entries(filterFeatures).length
            ? Object.entries(filterFeatures).map((feature, index) => {
              const title:string = feature[0] === 'parentAndRelevantChildCategories' ? 'Category' : feature[0] === 'allLanguages' ? 'Language' : 'Course Level';
              return (
                <FilteringContent
                  title={title}
                  key={index}
                  content={feature[1]}
                  filterFeature={title === 'Category' ? 'subCategoryIds' : title === 'Language' ? 'languageIds' : 'difficultyIds'}
                />
              );
            }) : null
        }
      </div>
      <button className={visibilityButton} type="button" onClick={toggleFilterTabVisibility}>{'<'}</button>
    </div>
  );
};
export default FilterComponent;
