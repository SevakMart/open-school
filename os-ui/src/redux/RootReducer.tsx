import { combineReducers } from 'redux';
import chooseSubcategoryReducer from './Slices/ChoosSubcategorySlice';
import userInfoReducer from './Slices/loginUserSlice';
import allLearningPathFilterParamsReducer from './Slices/AllLearningPathFilterParamsSlice';
import forgotPasswordEmailReducer from './Slices/ForgotPasswordEmailSlice';
import signedUpUserIdReducer from './Slices/SignedUpUserIdSlice';
import suggestedCourseReducer from './Slices/SuggestedCourseSlice';
import userEnrolledCourseByCourseStatusReducer from './Slices/UserEnrolledCourseByCourseStatus';
import allLearningPathCoursesReducer from './Slices/AllLearningPathCourseSlice';
import allMentorsFilterParamsReducer from './Slices/AllMentorsFilterParamsSlice';
import allMentorsSliceReducer from './Slices/AllMentorsSlice';
import savedMentorsReducer from './Slices/SavedMentorsSlice';
import savedCourseReducer from './Slices/SavedLearningPathSlice';
import homepageMentorListReducer from './Slices/HomepageMentorSlice';
import portalStatusReducer from './Slices/PortalOpenStatus';

const rootReducer = combineReducers({
  portalStatus: portalStatusReducer,
  userInfo: userInfoReducer,
  chooseSubcategories: chooseSubcategoryReducer,
  filterParams: allLearningPathFilterParamsReducer,
  forgotPasswordEmail: forgotPasswordEmailReducer,
  signedUpUserId: signedUpUserIdReducer,
  suggestedCourse: suggestedCourseReducer,
  userEnrolledCourseByCourseStatus: userEnrolledCourseByCourseStatusReducer,
  allLearningPathCourses: allLearningPathCoursesReducer,
  allMentorsFilterParams: allMentorsFilterParamsReducer,
  allMentorsList: allMentorsSliceReducer,
  savedMentors: savedMentorsReducer,
  savedCourse: savedCourseReducer,
  homepageMentorList: homepageMentorListReducer,
});
export default rootReducer;
