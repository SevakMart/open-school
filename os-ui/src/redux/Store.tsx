import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
