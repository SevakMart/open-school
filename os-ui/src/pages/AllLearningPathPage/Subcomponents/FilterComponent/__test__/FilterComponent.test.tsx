import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { PayloadAction } from '@reduxjs/toolkit';
import { store } from '../../../../../redux/Store';
import FilterComponent from '../FilterComponent';

jest.mock('redux-state-sync', () => ({
  createStateSyncMiddleware:
    () => () => (next: (action: PayloadAction) => void) => (action: PayloadAction) => next(action),
  initMessageListener: () => jest.fn(),
}));

describe('Create test cases for FilterComponent component', () => {
  test('Create a snapshot tets', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <FilterComponent />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
