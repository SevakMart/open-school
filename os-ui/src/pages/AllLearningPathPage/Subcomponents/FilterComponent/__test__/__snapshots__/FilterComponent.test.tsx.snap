// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create test cases for FilterComponent component Create a snapshot tets 1`] = `
<DocumentFragment>
  <div
    class="mainContainer"
    style="display: flex;"
  >
    <div
      class="filterMainContent"
    >
      <p
        class="mainTitle"
      >
        Filter
      </p>
    </div>
    <button
      class="visibilityButton"
      type="button"
    >
      &lt;
    </button>
  </div>
</DocumentFragment>
`;
